src/PrestaShopBundle/Form/Admin/Extension/DefaultEmptyDataExtension.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Extension;
+
+use PrestaShopBundle\Form\DataTransformer\DefaultEmptyDataTransformer;
+use stdClass;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Class DefaultEmptyDataExtension extends every form type with additional default_empty_data option.
```

Comment:
I see this class has a connection with DefaultEmptyDataTransformer but I fail to understand exactly the usecase it solves. I am sorry 😅  .

Could it be possible to explain it in the PR body and/or in PHPDoc?

Reply:
The use case is actually explained in details in the `DefaultEmptyDataTransformer` PHPDoc description that was previously added to the branch This extension is only a "shortcut" to use it more easily via the new options that can be used directly in the field definitions

See description of the transformer here https://github.com/PrestaShop/PrestaShop/blob/e4f7c9d0e1afd68dbf3ecc118713b3e88a869a2a/src/PrestaShopBundle/Form/DataTransformer/DefaultEmptyDataTransformer.php#L33 It brings feature that is not available with the existing options from Symfony (empty_data does not behave as we could expect ^^)

Do you think it's still worth describing it all in this PR's description?

Reply:
Now I understand, but DefaultEmptyDataTransformer is still hard to grasp when you dont know the context.

It's not because DefaultEmptyDataTransformer phpDoc is not good, it's because the topic is complex ;)

No need to modify the PR body